snippets/legv8.snippets.json

```diff
@@ -0,0 +1,223 @@
+      "LDUR $reg6, [SP, #0] // restore $reg6",
+      "LDUR $reg5, [SP, #8] // restore $reg5",
+      "LDUR $reg4, [SP, #16] // restore $reg4",
+      "LDUR $reg3, [SP, #24] // restore $reg3",
+      "LDUR $reg2, [SP, #32] // restore $reg2",
+      "LDUR $reg1, [SP, #40] // restore $reg1",
+      "LDUR LR, [SP, #48] // restore LR",
+      "ADDI SP, SP, #56  // restore previous space on stack"
+    ],
+    "description": "Use this to save 7 registers in the Stack, including the return address of the procedure (where the procedure was called)"
+  },
+  "stack manipulation with 7 registers": {
+    "prefix": "stk7",
+    "body": [
+      "// storing registers in stack before changing them",
+      "SUBI SP, SP, #64 // make room on stack (SP) for 2 registers",
+      "STUR LR, [SP, #56] // store return adress (LR) of procedure",
+      "STUR $reg1, [SP, #48] // store $reg1",
+      "STUR $reg2, [SP, #40] // store $reg2",
+      "STUR $reg3, [SP, #32] // store $reg3",
+      "STUR $reg4, [SP, #24] // store $reg4",
+      "STUR $reg5, [SP, #16] // store $reg5",
+      "STUR $reg6, [SP, #8] // store $reg6",
+      "STUR $reg7, [SP, #0] // store $reg7",
+      "// body of the procedure",
+      "$body",
+      "// restoring saved registers from stack",
+      "LDUR $reg7, [SP, #0] // restore $reg7",
+      "LDUR $reg6, [SP, #8] // restore $reg6",
+      "LDUR $reg5, [SP, #16] // restore $reg5",
+      "LDUR $reg4, [SP, #24] // restore $reg4",
+      "LDUR $reg3, [SP, #32] // restore $reg3",
+      "LDUR $reg2, [SP, #40] // restore $reg2",
+      "LDUR $reg1, [SP, #48] // restore $reg1",
+      "LDUR LR, [SP, #56] // restore LR",
+      "ADDI SP, SP, #64  // restore previous space on stack"
+    ],
+    "description": "Use this to save 8 registers in the Stack, including the return address of the procedure (where the procedure was called)"
+  },
+  "incremental for loop": {
+    "prefix": "for_increasing",
+    "body": [
+      "${1:loop name}_loop:",
+      "//comparing pointer to array length; if equal then exit",
+      "SUBS XZR, ${2:counter}, ${3:upper_limit}",
+      "B.GE ${1}_loop_exit",
+      "//loop body",
+      " ",
+      "// increment counter ($2) by $4",
+      "ADDI $3, $3, #${4:increment_amount}",
+      "B $1_loop",
+      "${1}_loop_exit:"
+    ],
+    "description": "Creates a for loop from counter to upper_limit. for (; counter < upper_limit; counter += increment_amount); Note: the counter and array size must be initialized before the loop."
+  },
+  "decremental for loop": {
+    "prefix": "for_decreasing",
+    "body": [
+      "${1:loop name}:",
+      "//comparing pointer to 0; if equal then exit",
+      "CBZ ${2:counter}, ${3:exit loop} ",
+      "//loop body",
+      " ",
+      "// decrement pointer",
+      "SUBI $2, $2, #${4:Imm}",
+      "B $1",
+      "${3}:"
+    ],
+    "description": "Creates a for loop from Counter to 0. Note: the counter must be initialized before the loop"
```

Comment:
In updated version, the C code is wrong. This code snippet does not initialize the counter. Check out what I put here in the description.

Reply:
Same thing for the increasing for loop.

Reply:
Sorry, Exactly which part is wrong? I added "Note: the counter must be initialized before the loop." in the end of the description just to be clear.

Reply:
- In the description, having the C code equivalent helps make what the for loop is doing clear.
- Using the same terminology as other loops for terms such as `decrement_amount`.
- Not requiring the user to type the loop exit label manually... it can be the same as the `loop_name`, but with `_exit` appended.

Reply:
Understood! The commit in [https://github.com/patrickdemers6/legv8-vscode/commit/6c75e85bb73e94d84a577ef15270743c9d1ca22a](https://github.com/patrickdemers6/legv8-vscode/commit/6c75e85bb73e94d84a577ef15270743c9d1ca22a) should fix it. Let me know if there`s anything I could improve or fix.